packages/go_router/lib/src/delegate.dart

```diff
@@ -44,13 +44,21 @@ class GoRouterDelegate extends RouterDelegate<RouteMatchList>
 
   final GlobalKey<NavigatorState> _key = GlobalKey<NavigatorState>();
 
+  /// The list of completers for the promises when pushing asynchronous routes.
+  final Map<String, Completer<dynamic>> completers =
```

Comment:
This should be private I think:

```suggestion
  final Map<String, Completer<dynamic>> _completers =
```